include/cache.php

```diff
@@ -84,9 +84,14 @@ public static function get($key) {
 		$memcache = self::memcache();
 		if (is_object($memcache)) {
 			// We fetch with the hostname as key to avoid problems with other applications
-			$value = $memcache->get(get_app()->get_hostname().":".$key);
-			if (!is_bool($value)) {
-				return unserialize($value);
+			$cached = $memcache->get(get_app()->get_hostname().":".$key);
+			$value = @unserialize($cached);
+
+			// Only return a value if the serialized value is valid.
+			// We also check if the db entry is a serialized
+			// boolean 'false' value (which we want to return).
+			if ($cached === 'b:0;' || $value !== false) {
+				return $value;
```

Comment:
Not sure what is required of me, but I can provide some general info about Memcache usage with PHP:
- You set any value with `Memcache::set()`. You don't have to serialize it beforehand.
- You get the stored value with `Memcache::get()` or `null` if the key wasn't set. You don't need to unserialize it, the calling function takes care of the data type.

A typical Memcache call usually resembles the following model, whether Memcache is activated or not:

`````php
function() {
  $key = 'something';

  if (!is_null($value = Memcache::get($key)) {
    return $value;
  }

  // Processing yielding $value

  Memcache::set($key, $value);

  return $value;
}